src/vec.rs

```diff
@@ -28,20 +34,41 @@ use core::{cmp::Ordering, fmt, hash, iter::FromIterator, mem::MaybeUninit, ops,
 /// }
 /// assert_eq!(*vec, [7, 1, 2, 3]);
 /// ```
-pub struct Vec<T, const N: usize> {
+pub struct Vec<T, const N: usize, A = u8> {
     // NOTE order is important for optimizations. the `len` first layout lets the compiler optimize
     // `new` to: reserve stack space and zero the first word. With the fields in the reverse order
     // the compiler optimizes `new` to `memclr`-ing the *entire* stack space, including the `buffer`
     // field which should be left uninitialized. Optimizations were last checked with Rust 1.60
     len: usize,
 
-    buffer: [MaybeUninit<T>; N],
+    storage: VecBuf<T, N, A>,
 }
 
-impl<T, const N: usize> Vec<T, N> {
+impl<T, const N: usize, A> VecBuf<T, N, A> {
     const ELEM: MaybeUninit<T> = MaybeUninit::uninit();
     const INIT: [MaybeUninit<T>; N] = [Self::ELEM; N]; // important for optimization of `new`
 
+    const fn new() -> Self {
+        Self {
+            alignment_field: [],
+            buffer: Self::INIT,
+        }
+    }
+
+    const fn as_ptr(&self) -> *const T {
```

Comment:
you can implement `Deref` and `DerefMut` into `[T; N]` so all the usual slice methods are instantly available, without you having to wrap them.

Reply:
> you can implement `Deref` and `DerefMut` into `[T; N]` so all the usual slice methods are instantly available, without you having to wrap them.

My main concern is that then the methods won't be `const`.

Reply:
Does this actually have to be `const`?